src/tbb/tls.h

```diff
@@ -41,7 +41,14 @@ class basic_tls {
         return pthread_key_create(&my_key, dtor);
     }
     int  destroy()      { return pthread_key_delete(my_key); }
-    void set( T value ) { pthread_setspecific(my_key, (void*)value); }
+    void set(T value) {
+        void *
+#if __TBB_GCC_VERSION >= 110100 && !__clang__ && !__INTEL_COMPILER
```

Comment:
Do we want to add this workaround for all future versions of GCC?

Reply:
I guess the warning is valid since non-pointer value is used for pointer type.
So behavior won't changed in future GCC release, but updated for double check in future GCC release.